baseplate/clients/sqlalchemy.py

```diff
@@ -123,6 +125,9 @@ def parse(
 Parameters = Optional[Union[Dict[str, Any], Sequence[Any]]]
 
 
+SAFE_TRACE_ID = re.compile("^[A-Za-z0-9_-]+$")
```

Comment:
🔕 I like this approach better than the one discussed in the transition doc: only do the sanitization locally, instead of doing that globally.

Reply:
I think they're complementary. I'm not suggesting that we have this sort of validation up front, but just that we state in the spec some basic bounds on what the trace IDs can be so that we're not violating the spec by doing any sanity checks. I think we can agree trace IDs will never be 3 megabyte gifs, so there's gotta be some inbetween on validation we can do that's not strict but not completely open.